cpp/dolfin/mesh/Mesh.h

```diff
@@ -90,14 +90,17 @@ class Mesh
   /// @param[in] ghost_mode The ghost mode
   /// @param[in] num_ghost_cells Number of ghost cells on this process
   ///                            (must be at end of list of cells)
+  /// @param[in] custom_permutation Mapping of DOLFIN/UFC vertex
```

Comment:
I don't follow what the map is. Is it order of input to DOLFIN ordering? And should 'vertex' be 'node'?

Reply:
This is a mistake by me. i followed the old `vtk_mapping` documentation, which was not correct.

It should be Mapping of DOLFIN/UFC node ordering to VTK/XDMF ordering.
For a CG-1 triangle, the VTK ordering is counter clockwise, where it counts vertices, then counter clockwise on edges. The UFC mapping  counts vertices in the same way, but the edges from non-intersection lexiographically. This means that the permutation_map={0,1,2,5,3,4}, which means that the 4th dof in VTK ordering, `permutation_map[4]` is the `5`th dof in the UFC ordering

Reply:
Let's spin this out into a free function.

Reply:
Isn't `vtk_mapping` already a free function?

Reply:
Yes, but it doesn't perform a permutation. We could have something like:
```
/// Convert input cell topology to DOLFIN cell ordering
Eigen::Array<std::int64_t, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor> 
  gmsh_to_dolfin_topology(const Eigen::Ref<....> topology, 
                          std::vector<int> permutation);
```

Reply:
Yes, I can have a look at that tomorrow.

Reply:
Suggested layout:
```
//-----------------------------------------------------------------------------
Eigen::Array<std::int64_t, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>
mesh::gmsh_to_dolfin_topology(
    Eigen::Array<std::int64_t, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>
        topology,
    std::vector<int> permutation)
{
  Eigen::Array<std::int64_t, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>
      cells_dolfin(topology.rows(), topology.cols());
  for (int c = 0; c < cells_dolfin.rows(); ++c)
  {
    for (int v = 0; v < cells_dolfin.cols(); ++v)
    {
      cells_dolfin(c, v) = topology(c, permutation[v]);
    }
  }
  return cells_dolfin;
}

//-----------------------------------------------------------------------------
```
Thus if input is a VTK conforming CG2 triangle , with ordering:
```
    #  *-----*-----*   3----6-----2
    #  |  \        |   | \        |
    #  |  \        |   |   \      |
    #  *     *     *   7     8    5
    #  |       \   |   |       \  |
    #  |         \ |   |        \ |
    #  *-----*-----*   0----4-----1
    cells = np.array([[0, 1, 3, 4, 8, 7], [1, 2, 3, 5, 6, 8]])
```
The call: 
 ` mesh::gmsh_to_dolfin_topology(cells, {0, 1, 2, 4, 5, 3});`
yields:
`new_cells = [[0,1,3,8,7,4],[1,2,3,6,8,5]]`
However, this means that we have to implement a inverse of `vtk_mapping` (with the name `dolfin_mapping`?)